alacritty_terminal/src/index.rs

```diff
@@ -30,6 +30,15 @@ pub enum Side {
     Right,
 }
 
+impl Side {
+    pub fn opposite(self) -> Self {
+        match self {
+            Side::Right => Side::Left,
+            Side::Left => Side::Right,
+        }
+    }
+}
```

Comment:
Just for fun (please feel free to ignore), this could be `impl Neg for Side` pretty nicely IMO.

Reply:
I think we're getting too close into JavaScript land here. It's not immediately obvious to me if `!test` or `!Side::Right` returns a side or a boolean. Meanwhile when I read `test.opposite()` I know more or less what's going on and I'm aware that to see what the function does I should do goto definition.

So I think while this might look smart and sleek initially, I think we'd shoot ourselves in the foot. Also it would be `Not`, not `Neg` I think.